theano/tensor/basic.py

```diff
@@ -4372,7 +4372,7 @@ def infer_shape(self, node, ishapes):
                     if ele == -1:
                         requ[i] = missing
             elif crit == 1:  # we reshape to -1
-                requ = [mul(*ishapes[0])]
+                requ = [mul(*ishapes[0])] if ishapes[0] else [1]
```

Comment:
mul() fails when there is no argument, i.e when ishapes[0] is empty. Then we give it a scalar dimension of 1.